doc/apidoc/classes.rst

```diff
@@ -45,6 +45,37 @@ Distributions
 .. autoclass:: qutip.QFunc
     :members:
 
+Environments
+------------
+
+.. autoclass:: qutip.core.BosonicEnvironment
+    :members:
+
+.. autoclass:: qutip.core.DrudeLorentzEnvironment
+    :members:
+    :inherited-members:
+    :show-inheritance:
+    :exclude-members: from_correlation_function, from_power_spectrum, from_spectral_density
```

Comment:
Small question -- why are these alternative constructors being excluded? Wouldn't users want to see them? Are they shown somwhere else perhaps?

Reply:
They are shown in the base class, `BosonicEnvironment`. It would not make much sense to call them on the specialized subclasses like `DrudeLorentzEnvironment`, where the correlation function, power spectrum and spectral density are all already determined. (Does that mean these alternative constructors should be `staticmethod` rather than `classmethod`?)

Reply:
Regarding the docs: Ah, I see.

Its traditional to make these alternative constructors class methods, because they usually return an new instance of the class they're attached to. The situation here is a bit odd because these helper methods return instances of sub-classes (`_BosonicEnvironment_fromSD` and friends). It's also odd because these methods will also appear on all the sub-classes too.

I don't think it's necessarily bad how it is, perhaps just a bit surprising.